### PR TITLE
Vcl has long long

### PR DESCRIPTION
**Pull request by hjmjohnson**

### CODE REVIEW

**Comment by hjmjohnson** [comment]
100% tests passed, 0 tests failed out of 1029
Total Test time (real) =  54.79 sec

All tests pass, even with the maximum number of options compiled.  

image

**Comment by hjmjohnson** [comment]
@dzenanz I'm about to start testing with ITK with the hope of updating ITK.

File: vcl/vcl_compiler.h.in [hunk 1]
```diff
@@ -1,95 +1,31 @@
...
-// Each each major release, define a release number
-//     #define VCL_XYZ_4
-// Avoid using the marketing name for the release number, because it's
-// harder to follow. For example, Microsoft Visual C++ .NET 2003 is
-// better called Visual C++ 7.
-// For each minor version, define the appropriate minor version number
-//     #define VCL_XYZ_40
-// If necessary, define the patchlevel too:
-//     #define VCL_XYZ_401
+// This file either is generated from vcl_compiler.h.in
```
Comment (dzenanz) [inline vcl/vcl_compiler.h.in]: Remove word `either`.

File: vcl/vcl_compiler.h.in [hunk 2]
```diff
@@ -218,6 +173,51 @@
...
+
+#if VXL_COMPILER_IS_GNU
+# if (__GNUC__ < 4)  //Feature support is now more finely tested in the vcl_compiler_detection.h file
+#  error "Invalid VXL_COMPILER_IS_GNU version not supported."
+# endif
+#endif
+
+#if VXL_COMPILER_IS_MSVC
+# if defined(_MSC_VER)
+#  if _MSC_VER < 1900     // Visual Studio 2015 = Version 14.x is needed to support
```
Comment (dzenanz) [inline vcl/vcl_compiler.h.in]: Remove comment

File: vcl/vcl_compiler.h.in [hunk 3]
```diff
@@ -218,6 +173,51 @@
...
+#   error "Invalid _MSC_VER version Visual Studio 2015 = Version 14.x is needed to support C++11 features"
```
Comment (dzenanz) [inline vcl/vcl_compiler.h.in]: Put a period (`.`) after version:
Invalid _MSC_VER version. Visual Studio 2015 = ...

**Pull request merged**